Src/PerformanceCollector/Shared/Implementation/IPerformanceCollector.cs

```diff
@@ -31,5 +30,12 @@ internal interface IPerformanceCollector
         /// <param name="error">Captures the error logged.</param>
         /// <param name="blockCounterWithInstancePlaceHolder">Boolean that controls the registry of the counter based on the availability of instance place holder.</param>
         void RegisterCounter(string perfCounter, string reportAs, bool isCustomCounter, out string error, bool blockCounterWithInstancePlaceHolder);
+
+        /// <summary>
+        /// Removes a counter.
+        /// </summary>
+        /// <param name="perfCounter">Name of the performance counter to remove.</param>
+        /// <param name="reportAs">ReportAs value of the counter to remove.</param>
+        void RemoveCounter(string perfCounter, string reportAs);
```

Comment:
Consider using snapshot approach for configuring. Instead of adding and removing counters - just drop the old collector and instantiate the new one. This way you'd ensure better consistency between what customer wanted to configure and what is actually configured